File: backend/app/schemas/flight.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class FlightOptionResponse(BaseModel):
    id: uuid.UUID
    airline_code: str
    airline_name: str
    flight_numbers: str
    origin_airport: str
    destination_airport: str
    departure_time: datetime
    arrival_time: datetime
    duration_minutes: int
    stops: int
    stop_airports: str | None
    price: float
    currency: str
    cabin_class: str | None
    seats_remaining: int | None
    is_alternate_airport: bool
    is_alternate_date: bool
    score: float | None = None

    model_config = {"from_attributes": True}

File: frontend/src/types/flight.ts
export interface FlightOption {
  id: string;
  airline_code: string;
  airline_name: string;
  flight_numbers: string;
  origin_airport: string;
  destination_airport: string;
  departure_time: string;
  arrival_time: string;
  duration_minutes: number;
  stops: number;
  stop_airports: string | null;
  price: number;
  currency: string;
  cabin_class: string | null;
  seats_remaining: number | null;
  is_alternate_airport: boolean;
  is_alternate_date: boolean;
  within_flexibility?: boolean;
  score?: number;
}
